src/CommunityToolkit.Aspire.RavenDB.Client/RavenDBClientExtension.cs

```diff
@@ -0,0 +1,251 @@
+
+        builder.Services.AddKeyedTransient<IAsyncDocumentSession>(serviceKey,
+            (sp, _) => sp.CreateAsyncDocumentSession(documentStore));
+    }
+
+    private static void AddHealthCheck(
+        this IHostApplicationBuilder builder,
+        string healthCheckName,
+        RavenDBSettings settings)
+    {
+        if (settings.DisableHealthChecks)
+            return;
+
+        builder.TryAddHealthCheck(
+            healthCheckName,
+            healthCheck => healthCheck.AddRavenDB(options =>
+                {
+                    options.Database = settings.DatabaseName;
+                    options.Urls = settings.Urls;
+                    options.Certificate = settings.GetCertificate();
+                },
+                healthCheckName,
+                null,
+                null,
+                settings.HealthCheckTimeout > 0 ? TimeSpan.FromMilliseconds(settings.HealthCheckTimeout.Value) : null));
+    }
+
+    private static IDocumentStore CreateRavenClient(RavenDBSettings ravenDbSettings)
+    {
+        var documentStore = new DocumentStore()
+        {
+            Urls = ravenDbSettings.Urls,
+            Database = ravenDbSettings.DatabaseName,
+            Certificate = ravenDbSettings.GetCertificate()
+        };
+
+        ravenDbSettings.ModifyDocumentStore?.Invoke(documentStore);
+
+        documentStore.Initialize();
+        
+        if (ravenDbSettings.CreateDatabase)
+        {
+            var databaseRecord = new DatabaseRecord(ravenDbSettings.DatabaseName);
+            documentStore.Maintenance.Server.Send(new CreateDatabaseOperation(databaseRecord));
+        }
+
+        return documentStore;
+    }
+
+    private static IDocumentSession CreateDocumentSession(this IServiceProvider provider,
+        IDocumentStore documentStore) => documentStore.OpenSession();
+
+    private static IAsyncDocumentSession CreateAsyncDocumentSession(this IServiceProvider provider,
+        IDocumentStore documentStore) => documentStore.OpenAsyncSession();
+
+    private static void ValidateSettings(
+        IHostApplicationBuilder builder,
+        RavenDBSettings settings)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(settings.Urls);
+
+        if (settings.Urls.Length == 0)
+            throw new InvalidDataException("At least one connection URL must be provided in 'RavenDBSettings.Urls'.");
+
+        if (settings.CreateDatabase && string.IsNullOrWhiteSpace(settings.DatabaseName))
+            throw new InvalidDataException("A database name must be specified in 'RavenDBSettings.DatabaseName' when 'RavenDBSettings.CreateDatabase' is set to true.");
+    }
+
+    private static void ValidateSettings(
+        IHostApplicationBuilder builder,
+        string[] connectionStrings)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(connectionStrings);
+
+        if (connectionStrings.Length == 0)
```

Comment:
we already checking that in `ValidateSettings(builder, settings)`

Reply:
removed